### ISSUE TITLE
Allow number after undercore in CamelCase rule

### ISSUE THREAD

**Issue by radarfox** [eslint/eslint#11530]
**What rule do you want to change?**

`camelcase`

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

new default behavior

**Please provide some example code that this change will affect:**

Examples of **correct** code for this rule:

```js
/*eslint camelcase: "error"*/

var foo_1 = true;
function foo_3_2 () {
  // ...
}
```

**What does the rule currently do for this code?**

Rule considers code as incorrect. Output of `eslint@5.15.3` for the example code:

```
3:5   error    Identifier 'foo_1' is not in camel case    camelcase
4:10  error    Identifier 'foo_3_2' is not in camel case  camelcase
```

**What will the rule do after it's changed?**

Rule should consider identifiers with number after underscore as a correct code.

My real world use-case is that I have an identifier that must contain semver version. For example, there is a platform called "webOS Signage 3.2", so I would like to use identifiers for classes like `KeyboardWebosSignage3_2` when implementing functionality for this platform. Separators are important in semver, if you omit them, then you can't distinguish between `1.1` and `11` (eleven). You can't use dots in JS identifiers, only `$` or `_`. I think that common practice is to replace the out of range characters with `_`. 

**Are you willing to submit a pull request to implement this change?**

I looked at the rule source and I can't find the location of the lowercase check for the next character after underscore, so probably not on my own.

**Comment by platinumazure** [comment]
I'm :-1: to this enhancement because you could use the [`id-match`](https://eslint.org/docs/rules/id-match) rule to cover this case.

**Comment by radarfox** [comment]
Alright, if this is an intended behavior, I'm closing this.

I ended up with using the following settings in config, if anyone interested: `"camelcase": ["error", {"allow": ["_\\d"]}]`
